cloud/scope/powervs_cluster_test.go

```diff
@@ -3043,3 +3045,958 @@ func TestCreateVPCSubnet(t *testing.T) {
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("Returns error when serviceInstance and VPC is not set in Status", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMVPCClient:         mockVPC,
+			ResourceClient:       mockResourceController,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					ResourceGroup: &infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("resourceGroupID")},
+					Zone:          ptr.To("us-east-1"),
+					VPC:           &infrav1beta2.VPCResourceReference{Region: ptr.To("region")},
+				},
+			},
+		}
+
+		err := clusterScope.createTransitGateway()
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("Fails to get transitGateway location and routing", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMVPCClient:         mockVPC,
+			ResourceClient:       mockResourceController,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					ResourceGroup: &infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("resourceGroupID")},
+					Zone:          ptr.To("zone-ID"),
+					VPC:           &infrav1beta2.VPCResourceReference{},
+				},
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					ServiceInstance: &infrav1beta2.ResourceReference{
+						ID: ptr.To("serviceInstanceID"),
+					},
+					VPC: &infrav1beta2.ResourceReference{
+						ID: ptr.To("vpcID"),
+					},
+				},
+			},
+		}
+
+		err := clusterScope.createTransitGateway()
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("Return error while creating transit Gateway", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMVPCClient:         mockVPC,
+			ResourceClient:       mockResourceController,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					ResourceGroup: &infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("resourceGroupID")},
+					Zone:          ptr.To("us-east-1"),
+					VPC:           &infrav1beta2.VPCResourceReference{Region: ptr.To("region")},
+				},
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					ServiceInstance: &infrav1beta2.ResourceReference{
+						ID: ptr.To("serviceInstanceID"),
+					},
+					VPC: &infrav1beta2.ResourceReference{
+						ID: ptr.To("vpcID"),
+					},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().CreateTransitGateway(gomock.Any()).Return(nil, nil, errors.New("failed to create transit Gateway"))
+		err := clusterScope.createTransitGateway()
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("creates transitGateway but return error while getting vpc details", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("creates transitGateway but return error while getting VPC details", func(t *testing.T) {
```
(usually we use capital letters for shortforms)

Reply:
also some places you are using Transit Gateway, transitGateway, TransitGateway, Lets keep it uniform across all the places.